File: CS/SpreadsheetPivotTableExamples/CodeUtils/SpreadsheetExampleEvaluatorByTimer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SpreadsheetPivotTableExamples
{
    #region SpreadsheetExampleEvaluatorByTimer
    public class SpreadsheetExampleEvaluatorByTimer : ExampleEvaluatorByTimer
    {
        public SpreadsheetExampleEvaluatorByTimer()
            : base()
        {
        }

        protected override ExampleCodeEvaluator GetExampleCodeEvaluator(ExampleLanguage language)
        {
            if (language == ExampleLanguage.VB)
                return new SpreadsheetVbExampleCodeEvaluator();
            return new SpreadsheetCSExampleCodeEvaluator();
        }
    }
    #endregion
}

File: VB/SpreadsheetPivotTableExamples/CodeUtils/SpreadsheetExampleEvaluatorByTimer.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace SpreadsheetPivotTableExamples
    #Region "SpreadsheetExampleEvaluatorByTimer"
    Public Class SpreadsheetExampleEvaluatorByTimer
        Inherits ExampleEvaluatorByTimer

        Public Sub New()
            MyBase.New()
        End Sub

        Protected Overrides Function GetExampleCodeEvaluator(ByVal language As ExampleLanguage) As ExampleCodeEvaluator
            If language = ExampleLanguage.VB Then
                Return New SpreadsheetVbExampleCodeEvaluator()
            End If
            Return New SpreadsheetCSExampleCodeEvaluator()
        End Function
    End Class
    #End Region
End Namespace
